File: src/main/java/yunas/router/Route.java
package yunas.router;

import yunas.Controller;
import yunas.http.DefaultContentType;
import yunas.http.HttpMethod;

/**
 * Route.
 *
 */
public class Route {


    private HttpMethod method;
    private String path;
    private Controller controller;
    private DefaultContentType defaultContentType;

    public Route(HttpMethod method, String path, Controller controller, DefaultContentType defaultContentType) {
        this.method = method;
        this.path = path;
        this.controller = controller;
        this.defaultContentType = defaultContentType;
    }

    public Controller getController() throws IllegalAccessException {

        try {
            return controller.getClass().newInstance();
        } catch (InstantiationException | IllegalAccessException e) {
            // When it is lambda in Kotlin.
            return controller;
        }

    }

    public HttpMethod getMethod() {
        return method;
    }

    public String getPath() {
        return path;
    }

    public DefaultContentType getDefaultContentType() {
        return defaultContentType;
    }
}

File: src/main/kotlin/yunas/router/Route.kt
package yunas.router

import yunas.Controller
import yunas.http.DefaultContentType
import yunas.http.HttpMethod

/**
 * Route.
 *
 */
class Route(val method: HttpMethod, val path: String, private val controller: Controller, val defaultContentType: DefaultContentType) {

    @Throws(IllegalAccessException::class)
    fun getController(): Controller {

        try {
            return controller.javaClass.newInstance()
        } catch (e: InstantiationException) {
            // When it is lambda in Kotlin.
            return controller
        } catch (e: IllegalAccessException) {
            return controller
        }

    }
}
